### PR TITLE
Make internal/test/daemon.Daemon swarm aware

### PR DESCRIPTION
**Pull request by vdemeester**
This remove the daemon.Swarm construction by make the new test Daemon
struct aware of swarm.

This is a follow-up for https://github.com/moby/moby/pull/36824. Next step will be to do some cleaning (on code calling this) and move more api helpers to the `internal/test/daemon` package :angel: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]
LGTM assuming CI is happy.

**Comment by cpuguy83** [comment]
```
18:36:17 internal/test/daemon/daemon.go:169:29:warning: t can be github.com/docker/docker/vendor/github.com/gotestyourself/gotestyourself/assert.TestingT (interfacer)
18:36:17 internal/test/daemon/daemon.go:167:1:warning: comment on exported method Daemon.NewClientT should be of the form "NewClientT ..." (golint)
18:36:17 internal/test/daemon/swarm.go:30:1:warning: comment on exported method Daemon.SwarmInit should be of the form "SwarmInit ..." (golint)
18:36:17 internal/test/daemon/swarm.go:54:1:warning: comment on exported method Daemon.SwarmLeave should be of the form "SwarmLeave ..." (golint)
18:36:17 internal/test/daemon/swarm.go:76:1:warning: comment on exported method Daemon.SwarmUnlock should be of the form "SwarmUnlock ..." (golint)
18:36:17 integration/internal/swarm/service.go:20:1:warning: dockerdBinary is unused (deadcode)
18:36:17 internal/test/daemon/swarm.go:12:1:warning: defaultSwarmListenAddr is unused (deadcode)
18:36:17 internal/test/daemon/swarm.go:12:1:warning: defaultSwarmPort is unused (deadcode)
18:36:17 internal/test/daemon/swarm.go:42:1:warning: comment on exported method Daemon.SwarmJoin should be of the form "SwarmJoin ..." (golint)
```

**Comment by vdemeester** [comment]
@cpuguy83 oups :sweat_smile: forgot to run `validate` before creating the PR :weary:

**Comment by cpuguy83** [comment]
Lots of swarm errors.

**Comment by vdemeester** [comment]
@cpuguy83 oh I see how I messed that up in my latest changes.. :sweat_smile:

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: integration-cli/daemon/daemon.go [hunk 1]
```diff
@@ -150,6 +148,21 @@ func (d *Daemon) WaitRun(contID string) error {
 	return WaitInspectWithArgs(d.dockerBinary, contID, "{{.State.Running}}", "true", 10*time.Second, args...)
 }
 
+// CmdRetryOutOfSequence tries the specified command against the current daemon for 10 times
+func (d *Daemon) CmdRetryOutOfSequence(args ...string) (string, error) {
```
Comment (thaJeztah) [inline integration-cli/daemon/daemon.go]: I think there’s a PR hat removes this, as it was no longer needed

Comment (vdemeester) [reply integration-cli/daemon/daemon.go]: ohh interesting :stuck_out_tongue_closed_eyes:

Comment (thaJeztah) [reply integration-cli/daemon/daemon.go]: Let's remove this again in a follow up :+1:

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**